prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -650,7 +683,13 @@ void rxr_pkt_handle_eager_rtm_send_completion(struct rxr_ep *ep,
 
 	tx_entry = (struct rxr_tx_entry *)pkt_entry->x_entry;
 	assert(tx_entry->total_len == rxr_pkt_req_data_size(pkt_entry));
-	rxr_cq_handle_tx_completion(ep, tx_entry);
+	/*
+	 * If applicaiton do not want RNR completion, we wrote TX completion
+	 * when the packet was sent
+	 */
+	if (rxr_ep_write_rnr_completion(ep))
```

Comment:
This seems really high chance of error.  Rather than running the same test twice, can we store a bit in the packet header to tell us whether or not to write a completion?  The choice logic should be in one place.

Reply:
In new revision, I added a commit to set `RXR_NO_COMPLETION` flag after completion has been written, and used that flag here.